store/store_test.go

```diff
@@ -369,7 +359,51 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryDetails(c *C) {
 	c.Check(result.Epoch, Equals, "0")
 
 	c.Check(repo.SuggestedCurrency(), Equals, "GBP")
-	c.Check(result.Private, Equals, true)
+
+	// skip this one until the store supports it
+	// c.Check(result.Private, Equals, true)
+
+	c.Check(snap.Validate(result), IsNil)
+}
+
+func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryDetailsDevmode(c *C) {
+	mockDevmodeJSON := strings.Replace(MockDetailsJSON, `"strict"`, `"devmode"`, -1)
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		c.Check(r.UserAgent(), Equals, userAgent)
+
+		c.Check(r.URL.Path, Equals, "/details/hello-world")
+
+		query := r.URL.Query()
+		c.Check(query.Get("channel"), Equals, "edge")
+
+		if query.Get("confinement") == "strict" {
+			w.WriteHeader(http.StatusNotFound)
+			io.WriteString(w, "{}")
+		} else {
+			w.Header().Set("X-Suggested-Currency", "GBP")
+			w.WriteHeader(http.StatusOK)
+			io.WriteString(w, mockDevmodeJSON)
+		}
+	}))
+
+	c.Assert(mockServer, NotNil)
+	defer mockServer.Close()
+
+	detailsURI, err := url.Parse(mockServer.URL + "/details/")
+	c.Assert(err, IsNil)
+	cfg := SnapUbuntuStoreConfig{
+		DetailsURI: detailsURI,
+	}
+	repo := NewUbuntuStoreSnapRepository(&cfg, "")
+	c.Assert(repo, NotNil)
+
+	// the actual test
+	result, err := repo.Snap("hello-world", "edge", false, nil)
```

Comment:
Hm, not releated to this branch (as such) but the `true`/`false` here trips me over so easily. I really whish we could use a flag here so that its `repo.Snap("hello-world", "edge", DevMode, nil)`